### PR TITLE
Option to remove DATA from reply option, fixed tests

### LINKED ISSUE #45: Failing tests
Hi larshp, 
I saw some tests are failling (7.51)
"test" test method on zcl_swag
"char" method on swag_map_type 

Overall the coverage is not that good too...

### PR DESCRIPTION
**Pull request by xinitrc86**
Hello Lars,
Good afternoon.
This is to fix issue #24, my idea was to be the least disruptive and keep syntax to old elements.
This also fix #45 (broken tests) and adds a new one for this feature.

Basically I added a new structure to the meta.

    TYPES:
      BEGIN OF sty_response,
        remove_data_object TYPE abap_bool,
      END OF sty_response.
    TYPES:
      BEGIN OF ty_meta,
        ...
        response_settings TYPE sty_response,
      END OF ty_meta .
And calling a handle_response on json_reply that will eventually:

  DATA lv_length TYPE i.
  DATA lv_minus_data TYPE i.
  lv_length = strlen( cv_data_as_string ).
  lv_minus_data = lv_length - 9.

  IF lv_minus_data <= 0.
    RETURN.
  ENDIF.
  "start has |{"DATA":| (8) end has |}| (1)
  cv_data_as_string = cv_data_as_string+8(lv_minus_data).
Regards,
Felipe

### CODE REVIEW

**Comment by xinitrc86** [comment]
This is the one ;)

File: src/zcl_swag.clas.testclasses.abap [hunk 1]
```diff
@@ -15,11 +15,18 @@ CLASS ltcl_swag DEFINITION FOR TESTING
...
+    DATA: mo_swag          TYPE REF TO zcl_swag,
+          mv_reply         TYPE string,
+          mo_dummy_handler TYPE REF TO zcl_dummy_swagger_handler.
 
     METHODS: setup,
-      test FOR TESTING RAISING cx_static_check.
+      given_a_foobar_meta
+        IMPORTING
+          iv_remove_data_object TYPE any OPTIONAL
+            PREFERRED PARAMETER iv_remove_data_object,
```
Comment (larshp) [inline src/zcl_swag.clas.testclasses.abap:26]: "PREFERRED PARAMETER iv_remove_data_object" can be removed

**Comment by larshp** [comment]
overall looks good to me, more options for the users are good, when its non-breaking

**Comment by larshp** [comment]
@xinitrc86 thanks, sorry about the delay, just ping me if I take too long time

if I merge this, do you have time for questions if someone runs into trouble?

**Comment by xinitrc86** [comment]
Hi Lars, 
Sure thing, specially if I'm mentioned ;)

Regards,
Felipe

**Comment by larshp** [comment]
thanks 👍

**Pull request merged**